### PR TITLE
Add missing wp block woocommerce classname

### PR DESCRIPTION
**Pull request by louwie17**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR updates the list of blocks below to include the `wp-block-<block name >` class on render:
- catalog-sorting
- product-average-rating
- product-button

Related to: https://github.com/woocommerce/woocommerce/issues/49739

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Tests for each block**
**Catalog sorting**
1. Create a new WooCommerce store and add some products, make sure to use a blockified theme
2. Go to the **Shop** page 
3. Find the **Default sorting** dropdown and inspect it.
4. Make sure the parent div ( `data-block-name="woocommerce/catalog-sorting"` ) includes the `wp-block-woocommerce-catalog-sorting` class name.

**product-average-rating**

1. Add a new page or edit an existing page and add a **Single product** block and select a product
2. Nested from the **single product** block add the **Product Average Rating (Beta)** block
3. Save the product and make sure to add at-least one review to the product.
4. View the page and inspect the average rating ( it should just be a number ). 
5. It should include the `wp-block-woocommerce-product-average-rating` class.

**Product Button**

1. Go to the **Shop** page
2. Inspect the **Add to cart** button
3. Make sure the parent block element (`data-block-name="woocommerce/product-button"`) contains the **wp-block-woocommerce-product-button**

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by mdperez86 — APPROVED** [review]
This is working as expected. The right class name is correctly added to each block.

Nice job @louwie17 !!

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php [hunk 1]
```diff
@@ -189,6 +191,21 @@ protected function render( $attributes, $content, $block ) {
...
 			';
 
+			$wrapper_attributes = get_block_wrapper_attributes(
+				array(
+					'class' => implode(
+						' ',
+						array_filter(
+							[
+								'wp-block-button wc-block-components-product-button',
+								esc_attr( $text_align_styles_and_classes['class'] ?? '' ),
```
Comment (gigitux) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php:201]: Is there a reason why we added this logic, even if `text_align_styles_and_classes` is never defined?

I noticed this by reviewing #50662

Comment (louwie17) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php:201]: Good catch 👍 that may of been a copy/paste mistake, I can spin up a separate PR to remove it.

Comment (gigitux) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php:201]: No worries @louwie17! I think that we can fix it in  https://github.com/woocommerce/woocommerce/pull/50662

Comment (louwie17) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php:201]: @gigitux I spun up a PR: https://github.com/woocommerce/woocommerce/pull/51729

**Pull request merged**